extra/mariabackup/xtrabackup.cc

```diff
@@ -4570,6 +4570,11 @@ static bool xtrabackup_backup_func()
 	}
 	msg("cd to %s", mysql_real_data_home);
 	encryption_plugin_backup_init(mysql_connection);
+	/* try to trigger a log checkpoint, but do not fail if the
+	server does not support this */
+	xb_mysql_query(mysql_connection,
```

Comment:
I think it makes sense to delay this call until after all other initialization is done. So that there is no minute-long delays in case of trivial parameter errors, e.g invalid path in --target-dir

Reply:
I agree, and I should have thought of this. Is there a way to send the statement and then wait for it to complete? I suppose that we would need another connection for that. It would be most convenient to do this from this current client thread. The execution of the statement needs to finish before `mariadb-backup --backup` reads the `ib_logfile0` of the server. I think that we need to send this statement as early as possible.

Reply:
One can send statement, then wait for it to complete, in 2 steps, and it does not need another connection This is [send](https://mariadb.com/kb/en/mysql_send_query/) , this is [wait](https://mariadb.com/kb/en/mysql_read_query_result/)

Reply:
279a5774a8d732514194207a05de3923a4ef6972 tries to achieve some overlap, which is not much, because the server’s `ib_logfile0` must be read very early on, in order to determine the latest checkpoint LSN.